repose-aggregator/services/service-client/impl/src/main/java/org/openrepose/core/services/serviceclient/akka/impl/AkkaServiceClientImpl.java

```diff
@@ -71,26 +74,49 @@ public AkkaServiceClientImpl(HttpClientService httpClientService) {
         quickFutureCache = CacheBuilder.newBuilder()
                 .expireAfterWrite(FUTURE_CACHE_TTL, FUTURE_CACHE_UNIT)
                 .build();
+    }
+
+    @PreDestroy
+    public void destroy() {
+        // tie this actor system to our bean lifecycle
+        actorSystem.shutdown();
+    }
 
+    @Override
+    public void setConnectionPoolId(String connectionPoolId) {
+        if (StringUtils.equals(this.connectionPoolId, connectionPoolId)) {
+            return;
+        }
 
-        tokenActorRef = actorSystem.actorOf(new Props(new UntypedActorFactory() {
-            public UntypedActor create() {
+        ActorRef oldTokenActorRef = tokenActorRef;
+        this.connectionPoolId = connectionPoolId;
+        createTokenActor(FORCE_CREATE);
 
-                return new AuthTokenFutureActor(serviceClient);
-            }
-        }).withRouter(new RoundRobinRouter(numberOfActors)), "authRequestRouter");
+        if (oldTokenActorRef != null) {
+            actorSystem.stop(oldTokenActorRef);
+        }
     }
 
+    private synchronized void createTokenActor(ForceCreate forceCreate) {
```

Comment:
Just always use the default connection pool ID, unless another is specified. 

Did the number of actors correspond to the connection pool size ever? What was the behavior before? Create an actor on each request? (Actors are super cheap, and you can create and throw them away.) If it's using a fixed actor pool size, the number of actors needs to at least be the size of the pool, so you can consume the entire pool. 

I'd recommend creating then a hierarchy of actors, so that an actor can control state below it, you can send a message to that actor to tell it to change it's connection pool. It would then send a kill message to the existing actors, and create a new set. The existing actors would finish up their work and then exit properly and any new work would go to the new actors with the new configuration.

OR: this might be a good time to refactor this. We really don't need actors here at all. All we need are futures, and an executor service.

https://github.com/rackerlabs/repose/blob/master/repose-aggregator/services/service-client/impl/src/main/java/org/openrepose/core/services/serviceclient/akka/impl/AuthTokenFutureActor.java

All this does is abstract a call so that's asynchronous. This can be done with the java Future constructs. It might make things a whole lot simpler, and it might be less work than trying to create a hierarchy of actors (which is how you're supposed to use akka to do what you're trying to do). It might also be a lot more work, I haven't wrapped my head around all the context.